DllImportGenerator/DllImportGenerator/TypeSymbolExtensions.cs

```diff
@@ -13,26 +13,34 @@ namespace Microsoft.Interop
 {
     static class TypeSymbolExtensions
     {
-        public static bool HasOnlyBlittableFields(this ITypeSymbol type)
+#pragma warning disable RS1024 // Compare symbols correctly. We are using the correct comparer, but the analyzer doesn't see it correctly.
+        public static bool HasOnlyBlittableFields(this ITypeSymbol type) => HasOnlyBlittableFields(type, new HashSet<ITypeSymbol>(SymbolEqualityComparer.Default));
+#pragma warning restore RS1024 // Compare symbols correctly
+
+        private static bool HasOnlyBlittableFields(this ITypeSymbol type, HashSet<ITypeSymbol> seenTypes)
```

Comment:
Do me a favor here and pass a recursion counter - when it reaches 0 we throw. Alternatively this could be converted to use a `Stack<T>` and turned into an iterative solution.

Reply:
Same protection for `IsConsideredBlittable`.

Reply:
As we discussed offline, I think we should leave Stack-overflow protection due to too many nested structs to the Roslyn compiler infrastructure.